drivers/storage/portworx/deployment.go

```diff
@@ -287,6 +287,10 @@ func (p *portworx) generateCloudStorageSpecs(
 		p.k8sClient,
 	}
 
+	if err := cloudStorageManager.CreateStorageDistributionMatrix(); err != nil {
+		logrus.Warnf("Failed to generate storage distribution matrix config map: %v", err)
```

Comment:
why is this just a warning?

Reply:
This is to not error out in scenarios where someone wants to create the config map out of band.

Reply:
If the config map does not exists, the pod creation will fail anyways as we cannot determine the node specific specs.

Reply:
Won't users create the config map with the same name?

Reply:
Yes they will have to. This function creates a config map. If it fails to do so we don't error out. 
The actual API that reads the config map returns an error if it can't find a config map.